lib/data/external/GCP/GcpApis/copyObject.js

```diff
@@ -0,0 +1,107 @@
+const async = require('async');
+const request = require('request');
+const { errors } = require('arsenal');
+
+const { logger, JsonError, checkSourceString } = require('../GcpUtils');
+const { logHelper } = require('../../utils');
+
+/**
+ * copyObject - minimum required functionality to perform object copy
+ * for GCP Backend
+ * @param {object} params - update metadata params
+ * @param {string} params.Bucket - bucket name
+ * @param {string} params.Key - object key
+ * @param {string} param.CopySource - source object
+ * @param {function} callback - callback function to call with the copy object
+ * result
+ * @return {undefined}
+ */
+function copyObject(params, callback) {
+    const { CopySource } = params;
+    if (!checkSourceString(CopySource)) {
+        if (typeof CopySource === 'string') {
+            return callback(errors.InvalidArgument);
+        }
+        return callback(errors.MissingParameter);
+    }
+    const sourceArray = CopySource.split('/');
+    const sourceBucket = sourceArray[0] ? sourceArray[0] : sourceArray[1];
+    const objectStart = sourceArray[0] ? 1 : 2;
+    const sourceObject =
+        sourceArray.slice(objectStart).reduce(
+            (prev, curr) => {
```

Comment:
You can use a functional expression instead of a true function for this: `(prev, curr) => prev ? ...`

Also, \``${curr}`\` <=> `curr`